pkg/backupdriver/backup_driver_controller_base.go

```diff
@@ -265,6 +280,14 @@ func NewBackupDriverController(
 		//DeleteFunc: rc.svcDeleteBackupRepositoryClaim,
 	}, resyncPeriod)
 
+	deleteSnapshotInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    func(obj interface{}) { ctrl.enqueueDeleteSnapshot(obj) },
+			DeleteFunc: func(obj interface{}) { ctrl.dequeDeleteSnapshot(obj) },
```

Comment:
We need UpdateFunc here. 

As we will filter out any non-New phase DeleteSnapshot CR, we need to handle UpdateFunc here. As the New phase of DeleteSnapshot CR was patched by UpdateStatus API which would trigger UpdateFunc rather than AddFunc.

Going forward, we can think about changing to filter out "empty" phase rather than "new" phase. That way, we don't need to handle UpdateFunc here in the EventHandler. We can do it together for all of Snapshot, CloneFromSnapshot and DeleteSnapshot CRD together.